### PR TITLE
Updated API wish Mesh Presence awareness and Power usage

### PR DESCRIPTION
**Pull request by mPlagge**
Added PowerUsage Presence and Mesh classes for API functionality. Also added example code for them.

### CODE REVIEW

File: include/Mesh.h [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#pragma once
+
+// #include <cstring>
+// #include <string.h>
+#include <microapp.h>
+// #include <stdio.h>
```
Comment (vliedel) [inline include/Mesh.h:6]: Remove commented out includes.

File: src/Mesh.cpp [hunk 2]
```diff
@@ -0,0 +1,67 @@
+#include <Mesh.h>
+
+uint8_t CMD_SIZE = 1;
```
Comment (vliedel) [inline src/Mesh.cpp:3]: Can be a static const.

File: src/Presence.cpp [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#include <Presence.h>
+
+#define MAX_ROOMS 64
```
Comment (vliedel) [inline src/Presence.cpp:3]: Can be a static const.

**Review by vliedel — CHANGES_REQUESTED** [review]
In general:
- Please change your code so it adheres to the [code style](https://github.com/crownstone/bluenet/blob/master/docs/CODE_STYLE.md)
- Rename, or split up api_examples.ino, so the name(s) tells what example it is. Or merge into the existing example.
- In retrospect, the mesh structs are not very handy, but we can refactor them later.

**Comment by mPlagge** [comment]
Thanks for your points of feedback. They are fixed in the latest commit.

**Review by vliedel — CHANGES_REQUESTED** [review]
Getting there =]
Needs documentation of the public methods, and style isn't done yet (especially variable names).

**Comment by mPlagge** [comment]
Changed the wrong info in the docstrings in the lastest commit.

**Comment by mrquincle** [comment]
@vliedel is this PR outdated? Can It be closed?

**Comment by vliedel** [comment]
No, this is outdated by now, and I think the API is already included (though differently).

**Pull request closed without merging**